cmd/registry/cmd/upload/bulk/discovery.go

```diff
@@ -77,6 +80,7 @@ type uploadDiscoveryTask struct {
 	versionID string
 	specID    string
 	contents  []byte
+	info      *DiscoveryInfo
```

Comment:
Why a pointer? The only piece of code that needs to mutate the underlying data in a DiscoveryInfo struct is the YAML parsing code.

Reply:
Memory efficiency? This avoids copying the struct. 

I can change parseDiscoveryInfo to take a *DiscoveryInfo argument instead of returning *DiscoveryInfo and make this a struct instead of a pointer.